packages/blockchain/src/index.ts

```diff
@@ -131,100 +142,149 @@ export default class Blockchain implements BlockchainInterface {
     }
 
     this._heads = {}
-    this._staleHeadBlock = false
-    this._staleHeads = []
 
     this._lock = new Semaphore(1)
-    this._initDone = false
+
+    if (opts.genesisBlock && !opts.genesisBlock.isGenesis()) {
+      throw 'supplied block is not a genesis block'
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    this.initPromise = this._init(opts.genesisBlock)
+  }
+
+  public static async create(opts: BlockchainOptions = {}) {
+    const blockchain = new Blockchain(opts)
+    await blockchain.initPromise!.catch((e) => {
+      throw e
+    })
+    return blockchain
   }
 
   /**
    * Returns an object with metadata about the Blockchain. It's defined for backwards compatibility.
    */
   get meta() {
     return {
-      rawHead: this._headHeader,
+      rawHead: this._headHeaderHash,
       heads: this._heads,
       genesis: this._genesis,
     }
   }
 
   /**
-   * Fetches the meta info about the blockchain from the db. Meta info contains
-   * hashes of the headerchain head, blockchain head, genesis block and iterator
-   * heads.
+   * This method is called in the constructor and either setups the DB or reads values from the DB and makes these available to the consumers of Blockchain.
```

Comment:
Please keep the length of comment lines roughly to the line length of the code blocks (I am actually wondering why this is not done by linting automatically?). This is otherwise not very well readable, on my MacBook Air it e.g. completely goes out of screen.

Reply:
I will manually lint those, thought it indeed would be done by linter as well

Reply:
(this applies to most of the comments added)

Reply:
So [printWidth](https://github.com/ethereumjs/ethereumjs-config/blob/master/packages/lint/prettier.config.js#L4) setting in the linting config is set to `100`, for orientation

Reply:
> I will manually lint those, thought it indeed would be done by linter as well

??? Can't you then just run the linter or is it not working for you locally?

Reply:
Just checked, if I run `npm run lint:fix` doesn't change anything here, I'd assume it would break up those comments.

Reply:
Seems like linter does not fix it. I have installed this Rewrap extension in VSCode which helps wrapping the comments. Wrapped comments now in 4db67e9